### PR TITLE
Collections API implementation (except patch)

### PR DESCRIPTION
**Pull request by nbs-rh**
# Pull Request

## Description

Please provide a clear and concise description of the changes in this PR.

**Related Issue(s)**:
- Fixes #(issue number)
- Relates to [#RHOAIENG-47887](https://issues.redhat.com/browse/RHOAIENG-47887)
## Type of Change

Please mark the relevant option(s):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Test improvements
- [ ] CI/CD improvements
- [ ] Other (please describe):

## Component(s) Affected

Please mark the relevant component(s):

- [X] API endpoints
- [ ] Evaluation executors
- [ ] MLFlow integration
- [ ] Collection management
- [ ] Kubernetes/OpenShift deployment
- [ ] Configuration system
- [ ] Monitoring/metrics
- [ ] Storage
- [ ] Documentation
- [ ] Tests
- [ ] CI/CD
- [ ] Other:

## Changes Made

Please describe the specific changes made:

### Added
- Implementation for collections API - create, get, list, update, delete and corresponding FVT tests.

## Testing

Please describe how you tested your changes:

### Test Coverage
- [ ] Unit tests added/updated
- [X] Integration tests added/updated
- [ ] Manual testing performed
- [ ] No tests needed (documentation/minor changes)

### Test Results
- [X] All existing tests pass
- [X] New tests pass
- [ ] Manual testing successful

**Manual Testing Details** (if applicable):
```
Describe manual testing steps and results:
1.
2.
3.
```

### Performance Impact
- [X] No performance impact
- [ ] Performance improvement
- [ ] Performance regression (justified and documented)
- [ ] Performance impact unknown

## Documentation

- [X] Code comments added/updated
- [ ] API documentation updated
- [ ] README updated
- [ ] CONTRIBUTING guide updated
- [ ] Examples/tutorials added/updated
- [ ] No documentation changes needed

## Breaking Changes

If this PR introduces breaking changes, please describe them and the migration path:

```
Describe breaking changes:
-
-

Migration path:
1.
2.
```

## Security Considerations

- [X] No security implications
- [ ] Security review needed
- [ ] Authentication/authorization changes
- [ ] Sensitive data handling changes
- [ ] Network security changes

Please describe any security implications:

## Deployment Considerations

- [X] No deployment changes needed
- [ ] Environment variables added/changed
- [ ] Configuration changes required
- [ ] Database migrations needed
- [ ] Kubernetes manifests updated
- [ ] New dependencies added

**Deployment Notes**:

## Screenshots/Evidence

If applicable, add screenshots or other evidence of the changes:

## Checklist

Please confirm the following before submitting:

### Code Quality
- [X] Code follows the project's style guidelines
- [X] Self-review of code has been performed
- [X] Code is properly commented
- [ ] No new linting warnings introduced

### Testing
- [X] Tests have been added for new functionality
- [X] All tests pass locally
- [ ] Test coverage is adequate
- [ ] No regression in existing functionality

### Documentation
- [ ] Documentation has been updated as needed
- [ ] API documentation reflects changes
- [ ] Examples updated if needed
- [ ] Breaking changes documented

### Review Readiness
- [ ] PR title is clear and descriptive
- [ ] PR description is complete
- [ ] Commit messages follow conventional commits format
- [ ] Branch is up to date with main
- [ ] Ready for review

## Additional Notes

Add any additional notes, context, or questions for reviewers:

---

**For Reviewers**: Please check that all automated checks pass before approving. Pay special attention to:
- Test coverage and quality
- API contract compatibility
- Security implications
- Performance impact
- Documentation completeness

## Summary by CodeRabbit

* **New Features**
  * Full collection management API including PATCH (JSON Patch) support; create, retrieve, update, delete, list with pagination.

* **Validation / Errors**
  * Stronger input validation (required name/description, non-empty benchmarks) and clearer 400 Bad Request responses.

* **Database / Storage**
  * Enhanced persistence with transactional reads/updates, driver-specific SQL paths, and added collection scope column.

* **Tests**
  * Extensive BDD coverage and test data for CRUD, patching, validation, pagination, and edge cases.

### CODE REVIEW

File: internal/storage/sql/collections.go [hunk 1]
```diff
@@ -10,21 +17,214 @@ import (
...
+	// Execute the DELETE query
+	_, err = s.exec(nil, deleteQuery, id)
+	if err != nil {
+		s.logger.Error("Failed to delete collection", "error", err, "id", id)
+		return se.NewServiceError(messages.DatabaseOperationFailed, "Type", "collection", "ResourceId", id, "Error", err.Error())
+	}
+
+	s.logger.Info("Deleted collection", "id", id)
+
 	return nil
```
[the remark this answers is not in the document]
Comment (mariusdanciu) [reply internal/storage/sql/collections.go]: If the collection does not exist we should return 404

File: internal/storage/sql/collections.go [hunk 2]
```diff
@@ -10,21 +17,214 @@ import (
...
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, se.NewServiceError(messages.ResourceNotFound, "Type", "collection", "ResourceId", id)
+		}
+		// For now we differentiate between no rows found and other errors but this might be confusing
+		s.logger.Error("Failed to get collection", "error", err, "id", id)
+		return nil, se.NewServiceError(messages.DatabaseOperationFailed, "Type", "collection", "ResourceId", id, "Error", err.Error())
+	}
+
+	// Unmarshal the entity JSON into EvaluationJobConfig
```
[the remark this answers is not in the document]
Comment (mariusdanciu) [reply internal/storage/sql/collections.go]: Correct typo
